python/tank/bootstrap/manager.py

```diff
@@ -23,6 +23,9 @@ class ToolkitManager(object):
     and installations.
     """
 
+    # Maximum progress step number that can be passed to a progress callback.
+    MAX_PROGRESS_STEP_NUMBER = 8  # "Launching Engine..."
```

Comment:
I wonder if instead of using integers here, we could report back progress as a value from 0-1 representing the percentage of work complete? That would allow us to insert new steps without changing the max steps or modifying the step value for all calls to report progress. What do you think? Would that work for the client code?